packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -396,6 +397,88 @@ abstract class ProjectBase extends Resource implements IProject {
   }
 }
 
+/**
+ * Statuses for Log Configurations
+ */
+export enum LogStatus {
+  /**
+   * Enables the log configuration
+   */
+  ENABLED='ENABLED',
+
+  /**
+   * Disables the log configuration
+   */
+  DISABLED='DISABLED'
+}
+
+/**
+ * Information about logs built to an S3 bucket for a build project.
+ */
+export interface S3LogsConfig {
+  /**
+   * Encrypt the S3 build log output
+   * @default true
+   */
+  readonly encrypted?: boolean;
+
+  /**
+   * The S3 Bucket to send logs to
+   */
+  readonly bucket: s3.IBucket;
+
+  /**
+   * The path prefix for S3 logs
+   * @default - no prefix
+   */
+  readonly prefix?: string;
+
+  /**
+   * The current status of the logs in Amazon CloudWatch Logs for a build project
+   * @default ENABLED
+   */
+  readonly status?: LogStatus;
+}
+
+/**
+ * Information about logs built to an S3 bucket for a build project.
+ */
+export interface CloudwatchLogsConfig {
+  /**
+   * The S3 Bucket to send logs to
+   */
+  readonly logGroup: logs.ILogGroup;
+
+  /**
+   * The prefix of the stream name of the Amazon CloudWatch Logs
+   * @default - no prefix
+   */
+  readonly prefix?: string;
+
+  /**
+   * The current status of the logs in Amazon CloudWatch Logs for a build project
+   * @default ENABLED
+   */
+  readonly status?: LogStatus;
+}
+
+/**
+ * Information about logs for the build project. A project can create logs in Amazon CloudWatch Logs, an S3 bucket, or both.
+ */
+export interface LogsConfig {
```

Comment:
I understand you basically implemented the CloudFormation schema for this, but the CDK has a nicer way to implement this pattern. We call it the "union-like class".

It basically looks like this:

```ts
export interface ProjectLogsConfig {
  readonly logsConfiProperty: CfnProject.LogsConfigProperty;
}

export interface S3ProjectLogsOptions {
  readonly bucket: s3.IBucket;
  readonly encrypted?: boolean;
  readonly prefix?: string;
}

export interface CloudwatchProjectLogsOptions {
  readonly logGroup: logs.ILogGroup;
  readonly prefix?: string;
}

export abstract class ProjectLogs {
  public static s3(options: S3ProjectLogsOptions): ProjectLogs {
    return new S3ProjectLogsConfig(options);
  }

  public static cloudWatch(options: CloudwatchProjectLogsOptions): ProjectLogs {
    return new CloudwatchProjectLogs(optios);
  }

  // this is the method you call from the Project class
  public abstract bind(scope: Construct): ProjectLogsConfig;
}

class S3ProjectLogsConfig extends ProjectLogs {
  // ...
}

class CloudwatchProjectLogs extends ProjectLogs {
  // ...
}
```

Take a look at the `Source` and `Artifacts` classes for some inspiration, they also use this pattern.

Reply:
I'll get this updated. Thanks!